addons/sale_loyalty/models/sale_order.py

```diff
@@ -1339,8 +1339,9 @@ def _try_apply_program(self, program, coupon=None):
         elif program in self._get_applied_programs():
             return {'error': _("This program is already applied to this order."), 'already_applied': True}
         elif program.reward_ids:
-            global_reward = program.reward_ids.filtered('is_global_discount')
             applied_global_reward = self._get_applied_global_discount()
+            global_rewards = program.reward_ids.filtered('is_global_discount')
+            global_reward = global_rewards and max(global_rewards, key=lambda r: self._discountable_order(r)[0])
```

Comment:
```suggestion
            global_rewards = program.reward_ids.filtered('is_global_discount')
            global_reward = global_rewards and max(global_rewards, key=lambda r: self._discountable_order(r)[0])
            applied_global_reward = self._get_applied_global_discount()
```

Improve steps to produce(it is not clear and difficult)
- Install modules sale_management and sale_loyalty.
- Create two new Discount & Loyalty with the program type as Coupons and generate coupons
- In second Discount & Loyalty add multiple rewards
- Create sale order and apply coupon from first Discount & Loyalty and again try same with second one

Re-target branch to 18.0